packages/manager/apps/hpc-vmware-public-vcf-aas/src/components/Fields/index.tsx

```diff
@@ -1,33 +1,45 @@
 import React, { useId } from 'react';
 
 import clsx from 'clsx';
-import { Control, FieldValues, UseControllerProps, useController } from 'react-hook-form';
+import {
+  Control,
+  FieldPath,
+  FieldValues,
+  UseControllerProps,
+  useController,
+} from 'react-hook-form';
 
 import { OdsFormField, OdsText } from '@ovhcloud/ods-components/react';
 
-import { RhfFieldContext, useRhfFieldContext } from './RhfField.context';
+import { RhfFieldContext, RhfFieldContextParams, useRhfFieldContext } from './RhfField.context';
 import { RhfInput } from './RhfInput.component';
 import { RhfQuantity } from './RhfQuantity.component';
 
-type RhfFieldProps = React.ComponentProps<typeof OdsFormField> & {
-  controllerParams: UseControllerProps<FieldValues, string>;
+type RhfFieldProps<
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
```

Comment:
This type is already complicated, let's not complicated it more.
The ` = FieldValues` part specifies a default value in case the type is used without giving a value for the generic, but this won't be possible on the `RhfField` component anyway:
```suggestion
  TFieldValues extends FieldValues,
  TName extends FieldPath<TFieldValues>,
```